src/Microsoft.IdentityModel.Abstractions/Result.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+#nullable enable
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Represents a result that can be either successful or unsuccessful.
+    /// </summary>
+    /// <typeparam name="TResult"></typeparam>
+    /// <typeparam name="TError"></typeparam>
+    public readonly struct Result<TResult, TError> : IEquatable<Result<TResult, TError>>
```

Comment:
I'm not sure we want a public type simply called `Result`

Reply:
I am looking for an alternative name. If you can come up with a suggestion, I’ll happily take it.

Should be mentioned in the description, I believe.